internal/active/active.go

```diff
@@ -39,14 +59,37 @@ func CheckIfNodeActive(node models.Node, repos *repositories.Repos) (bool, error
 		metrics.FinalizedBlockHeight <= (latestBlockMetrics.FinalizedBlockHeight - AllowedBlocksBehind) {
 		log.Debugf(
 			"Node %s not active as metrics check failed. " +
-			"Node metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d] " +
-			"Best pool metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d]",
-			node.ID,
+				"Node metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d] " +
+				"Best pool metrics: BestBlockHeight[%d], FinalizedBlockHeight[%d]",
+			nodeID,
 			metrics.BestBlockHeight, metrics.FinalizedBlockHeight,
 			latestBlockMetrics.BestBlockHeight, latestBlockMetrics.FinalizedBlockHeight,
 		)
 		return false, nil
 	}
-	
+
 	return true, nil
 }
+
+// ActivateNodeIfReady sets node to active nodes if latest metrics are valid and node is not penalized
+func ActivateNodeIfReady(nodeID string, repos repositories.Repos) error {
+	node, err := repos.NodeRepo.FindByID(nodeID)
+	if err != nil {
+		return err
+	}
+
+	if node.Cooldown != 0 {
+		return nil
+	}
+
+	metricsValid, err := CheckIfMetricsValid(nodeID, &repos)
+	if err != nil {
+		return err
+	}
+
+	if metricsValid {
+		_ = repos.NodeRepo.AddNodeToActive(*node)
```

Comment:
Log?

Reply:
Both if it fails and if we add to active

Reply:
:+1: